### PR TITLE
Add messages to all uses of common-utils assert

### LINKED ISSUE #4439: Add messages to all asserts
Every call to `assert` or one of its functions in our code should have a message. We should simply modify our assert function in our [assert.ts](https://github.com/microsoft/FluidFramework/blob/716eb9e1b7c90ab4e1b1fa4f6ce78ed13e3b6d9b/common/lib/common-utils/src/assert.ts#L12) to require message.

There have been two errors in two days that I've personally investigated where the lack of message caused trouble, since the resulting error message in the logs is something like `false == true` or `undefined == true`. So this can conflate unrelated callstacks, or simply obscure the failing line of code.

### PR DESCRIPTION
**Pull request by ksbrar**
Addresses https://github.com/microsoft/FluidFramework/issues/4439, i.e. adds a message to every use of `assert` that relies on the browser-friendly version from the `common-utils` package.

I tried making useful messages, but would appreciate feedback!

### Other changes:

Along the way, I also noticed a couple uses of the common-utils assert in test code:
* [packages/drivers/odsp-driver/src/test/rateLimiter.spec.ts](https://github.com/microsoft/FluidFramework/pull/5367/files#diff-bb0ffc8dfdeb235ff3bd7c38793a3d2131a853fc247b5b8769ad73445097bd23) (switched this one to use typical `{ strict as assert }`)
* [packages/loader/driver-utils/src/test/blobAggregation.spec.ts ](https://github.com/microsoft/FluidFramework/pull/5367/files#diff-4ac11803764cfe844c150399c30523773330af83a89f88bc2e4beb9ab713ff09) (switches this one as well)

Also fixed one comment typo in [mergetree](https://github.com/microsoft/FluidFramework/pull/5367/files#diff-44376f4041a9802657345f8b3f961115e219b0c831da2b8f22894cf95d234bad)

### CODE REVIEW

**Review by anthony-murphy — APPROVED** [review]
this looks good to me. the only bummer in the bundle size increase, but i think it's a cost we have to pay inorder to have debuggable and diagnosable asserts.

**Comment by anthony-murphy** [comment]
### My one feedback. Please do smaller PRs in the future. this could easily have been done per packager, or per layer

File: examples/data-objects/prosemirror/src/fluidBridge.ts [hunk 1]
```diff
@@ -142,12 +142,13 @@ export class ProseMirrorTransactionBuilder {
...
 
-                assert(i < this.things.length);
+                assert(i < this.things.length, "Trying to insert removal node out-of-bounds!");
 
                 i = this.splitAt(position, i);
                 let length = range.segment.cachedLength;
                 while (length > 0) {
-                    assert(this.things[i].type === "ether");
+                    assert(this.things[i].type === "ether",
+                        `Current thing does not have 'ether' node type! Has '${this.things[i].type}' type.`);
```
Comment (anthony-murphy) [inline examples/data-objects/prosemirror/src/fluidBridge.ts:151]: don't put varialbes in assert messages. it makes them really hard to query. if we need variable, we should log or throw an error with the varialbe as a property

File: packages/dds/matrix/src/matrix.ts [hunk 2]
```diff
@@ -557,7 +561,8 @@ export class SharedMatrix<T extends Serializable = Serializable>
                 this.rows.applyMsg(msg);
                 break;
             default: {
-                assert(contents.type === MatrixOp.set);
+                assert(contents.type === MatrixOp.set,
+                    `SharedMatrix message contents have unexpected type: '${contents.type}'!`);
```
Comment (anthony-murphy) [inline packages/dds/matrix/src/matrix.ts:565]: no varialbes in asserts

File: examples/data-objects/prosemirror/src/fluidBridge.ts [hunk 3]
```diff
@@ -185,12 +186,13 @@ export class ProseMirrorTransactionBuilder {
...
 
-                assert(i < this.things.length);
+                assert(i < this.things.length, "Trying to insert annotations field on out-of-bounds node!");
 
                 i = this.splitAt(position, i);
                 let length = range.segment.cachedLength;
                 while (length > 0) {
-                    assert(this.things[i].type === "ether");
+                    assert(this.things[i].type === "ether",
+                        `Current thing does not have 'ether' node type! Has '${this.things[i].type}' type.`);
```
Comment (anthony-murphy) [inline examples/data-objects/prosemirror/src/fluidBridge.ts:195]: no varialbes in asserts

File: examples/data-objects/prosemirror/src/fluidCollabManager.ts [hunk 4]
```diff
@@ -117,7 +117,8 @@ export class FluidCollabManager extends EventEmitter implements IRichTextEditor
 
                     case ReferenceType.NestEnd:
                         const popped = nodeStack.pop();
-                        assert(popped!.type === nodeType);
+                        assert(popped!.type === nodeType,
+                            `NestEnd top-node type has wrong type: '${popped!.type}'!`);
```
Comment (anthony-murphy) [inline examples/data-objects/prosemirror/src/fluidCollabManager.ts:121]: no varialbes in asserts

File: examples/data-objects/table-view/src/grid.ts [hunk 5]
```diff
@@ -291,7 +291,8 @@ export class GridView {
 
             // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
             this.tdText = newParent.firstChild!;
-            console.assert(this.tdText.nodeType === Node.TEXT_NODE);
+            console.assert(this.tdText.nodeType === Node.TEXT_NODE,
+                `TableData text has wrong node type: '${this.tdText.nodeType}'!`);
```
Comment (anthony-murphy) [inline examples/data-objects/table-view/src/grid.ts:295]: no varialbes in asserts

File: examples/data-objects/webflow/src/html/formatters.ts [hunk 6]
```diff
@@ -111,7 +111,8 @@ export class InclusionFormatter extends Formatter<IInclusionState> {
     }
 
     public visit(layout: Layout, state: Readonly<IInclusionState>) {
-        assert(getDocSegmentKind(layout.segment) === DocSegmentKind.inclusion);
+        assert(getDocSegmentKind(layout.segment) === DocSegmentKind.inclusion,
+            `Layout segment has unexpected kind: '${getDocSegmentKind(layout.segment)}'!`);
```
Comment (anthony-murphy) [inline examples/data-objects/webflow/src/html/formatters.ts:115]: no variables in asserts

File: packages/dds/matrix/src/undoprovider.ts [hunk 7]
```diff
@@ -46,7 +46,8 @@ export class VectorUndoProvider {
 
             // For SharedMatrix, each IRevertibles always holds a single row/col operation.
             // Therefore, 'currentOp' must either be undefined or equal to the current op.
-            assert(this.currentOp === undefined || this.currentOp === operation);
+            assert(this.currentOp === undefined || this.currentOp === operation,
+                `On vector undo, unexpected 'currentOp' type/state: '${this.currentOp}'!`);
```
Comment (anthony-murphy) [inline packages/dds/matrix/src/undoprovider.ts:50]: no variables in asserts

File: packages/dds/merge-tree/src/client.ts [hunk 8]
```diff
@@ -332,12 +332,13 @@ export class Client {
...
-     * Performs the annotate based on the provided op
+     * Performs the remove based on the provided op
      * @param opArgs - The ops args for the op
-     * @returns True if the annotate was applied. False if it could not be.
+     * @returns True if the remove was applied. False if it could not be.
      */
     private applyRemoveRangeOp(opArgs: IMergeTreeDeltaOpArgs): boolean {
-        assert(opArgs.op.type === ops.MergeTreeDeltaType.REMOVE);
+        assert(opArgs.op.type === ops.MergeTreeDeltaType.REMOVE,
+            `Unexpected op type on range remove: '${opArgs.op.type}'!`);
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/client.ts:341]: no variables in asserts

File: packages/dds/merge-tree/src/client.ts [hunk 9]
```diff
@@ -375,7 +376,8 @@ export class Client {
      * @returns True if the annotate was applied. False if it could not be.
      */
     private applyAnnotateRangeOp(opArgs: IMergeTreeDeltaOpArgs): boolean {
-        assert(opArgs.op.type === ops.MergeTreeDeltaType.ANNOTATE);
+        assert(opArgs.op.type === ops.MergeTreeDeltaType.ANNOTATE,
+            `Unexpected op type on range annotate: '${opArgs.op.type}'!`);
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/client.ts:380]: no variables in asserts

File: packages/dds/merge-tree/src/client.ts [hunk 10]
```diff
@@ -410,7 +412,8 @@ export class Client {
      * @returns True if the insert was applied. False if it could not be.
      */
     private applyInsertOp(opArgs: IMergeTreeDeltaOpArgs): boolean {
-        assert(opArgs.op.type === ops.MergeTreeDeltaType.INSERT);
+        assert(opArgs.op.type === ops.MergeTreeDeltaType.INSERT,
+            `Unexpected op type on range insert: '${opArgs.op.type}'!`);
```
Comment (anthony-murphy) [inline packages/dds/merge-tree/src/client.ts:416]: no variables in asserts

File: packages/runtime/test-runtime-utils/src/mocks.ts [hunk 11]
```diff
@@ -612,7 +613,8 @@ export class MockSharedObjectServices implements IChannelServices {
     public static createFromSummary(summaryTree: ISummaryTree) {
         const contents: { [key: string]: string } = {};
         for (const [key, value] of Object.entries(summaryTree.tree)) {
-            assert(value.type === SummaryType.Blob);
+            assert(value.type === SummaryType.Blob,
+                `Unexpected summary type on mock createFromSummary: ${value.type}`);
```
Comment (anthony-murphy) [inline packages/runtime/test-runtime-utils/src/mocks.ts:617]: no variables in asserts

File: packages/loader/driver-utils/src/test/blobAggregation.spec.ts [hunk 12]
```diff
@@ -155,27 +159,31 @@ describe("BlobAggregationStorage", () => {
...
+            `Unexpected blob storage size: ${storage.blobs.size} vs expected 3`);
     });
 
     it("aggregation above 40 bytes only", async () => {
         const { storage } = await prep(true, 40);
 
         // Should skip one blob that is bigger than 40 bytes.
-        assert(storage.blobs.size === 4);
+        assert(storage.blobs.size === 4,
+            `Unexpected blob storage size: ${storage.blobs.size} vs expected 4`);
```
Comment (anthony-murphy) [inline packages/loader/driver-utils/src/test/blobAggregation.spec.ts:187]: no variables in asserts

File: packages/loader/driver-utils/src/buildSnapshotTree.ts [hunk 13]
```diff
@@ -49,7 +49,7 @@ function flattenCore(
             };
             entries.push(entry);
         } else {
-            assert(treeEntry.type === TreeEntry.Tree);
+            assert(treeEntry.type === TreeEntry.Tree, `Unexpected tree entry type on flatten: ${treeEntry.type}`);
```
Comment (anthony-murphy) [inline packages/loader/driver-utils/src/buildSnapshotTree.ts:52]: no variables in asserts

File: packages/drivers/debugger/src/sanitizer.ts [hunk 14]
```diff
@@ -447,7 +448,7 @@ export class Sanitizer {
      * @param contents - contents object to fix
      */
     fixAttachContents(contents: any): any {
-        assert(typeof contents === "object");
+        assert(typeof contents === "object", `Unexpected type on contents for fix of an attach: '${typeof contents}'.`);
```
Comment (anthony-murphy) [inline packages/drivers/debugger/src/sanitizer.ts:451]: no variables in asserts

**Review by markfields — APPROVED** [review]
🎂 Looks good, and 0.36 has been cut so I think this is ok to go in now, we will continue the conversation about bundle size, tagging, etc and if anything needs to happen before 0.37 releases we'll see to it.

**Comment by anthony-murphy** [comment]
i'm going to commit this and fast follow with a short code implementing, that I'm currenting prototyping. I'll likely checkin a pr to move short codes before the automation. current prototype takes less than 10 sec to run

**Pull request merged**